### PR TITLE
Quality arguments to `Accept` header should default to 1.0

### PR DESCRIPTION
**Pull request by hiddewie**
This PR ensures that Accept headers with mime types without a given quality value, are parsed as having a quality value of `1.0`.

The current value for mime types without a quality value is 0.0 (`nil.to_f`).

From the HTTP specification (RFC 9110), https://www.rfc-editor.org/rfc/rfc9110.html#name-accept and https://www.rfc-editor.org/rfc/rfc9110.html#name-quality-values:

> The weight is normalized to a real number in the range 0 through 1, where 0.001 is the least preferred and 1 is the most preferred; a value of 0 means "not acceptable". If no "q" parameter is present, the default weight is 1.

### CODE REVIEW

File: lib/grape/middleware/formatter.rb [hunk 1]
```diff
@@ -171,7 +171,7 @@ def mime_array
         vendor_prefix_pattern = /vnd\.[^+]+\+/
 
         accept.scan(accept_into_mime_and_quality)
-              .sort_by { |_, quality_preference| -quality_preference.to_f }
+              .sort_by { |_, quality_preference| -(quality_preference || '1.0').to_f }
```
Comment (dblock) [inline lib/grape/middleware/formatter.rb:174]: Does it matter that it doesn't work with `quality_preference = ''`:

```ruby
3.1.2 :012 > quality_preference = ''
 => "" 
3.1.2 :013 > (quality_preference || '1.0').to_f
 => 0.0 
```

If it's really about `nil` (and not an explicitly passed in 0.0), then I would just do this to avoid parsing the 1.0 float every time.

```
quality_preference ? quality_preference.to_f : 1.0
```

**Review by dblock — CHANGES_REQUESTED** [review]

**Review by dblock — COMMENTED** [review]
Add a test that with `nil`, `0.0`, and `invalid`, too?

File: lib/grape/middleware/formatter.rb (hunk 1 above)
Comment (hiddewie) [reply lib/grape/middleware/formatter.rb]: Yes, sounds good!

**Comment by hiddewie** [comment]
Thanks for the review. I changed the parsing to include all `q=...` values. This will try to parse every value to a floating point number.  Invalid (textual) values will be parsed as 0.0. The empty `q` value will become a default quality of 1.0 which looks like a fine behaviour for me.

**Comment by dblock** [comment]
Add to CHANGELOG and this is good to go!

**Comment by hiddewie** [comment]
Thanks, done!

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 #### Fixes
 
 * [#2364](https://github.com/ruby-grape/grape/pull/2364): Add missing requires - [@ericproulx](https://github.com/ericproulx).
+* [#2366](https://github.com/ruby-grape/grape/pull/2366): Quality arguments to `accept` header should default to 1.0 - [@hiddewie](https://github.com/hiddewie).
```
Comment (dblock) [inline CHANGELOG.md:14]: Let's say "Default quality to 1.0 in the Accept header when omitted".

**Review by dblock — COMMENTED** [review]
One more, should we also document this in README? We don't need to explain the entire purpose of `q`, but maybe call it out in https://github.com/ruby-grape/grape#header with an example that chooses JSON over XML? Maybe something like this?

"Grape will evaluate the relative quality preference included in Accept headers and default to a quality of 1.0 when omitted. In the following example a Grape API that supports XML and JSON in that order will return JSON."

```
curl ... -H "Accept: text/xml;q=0.8, application/json;q=0.9"
```

**Comment by hiddewie** [comment]
Sounds good, done!

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Merged, nice work, thanks for hanging in here with me!
